junit/src/main/java/org/kaazing/k3po/junit/rules/ScriptRunner.java

```diff
@@ -137,12 +139,19 @@ public ScriptPair call() throws Exception {
                     case FINISHED:
                         FinishedEvent finished = (FinishedEvent) event;
                         // note: observed script is possibly incomplete
-                        String observedScript = finished.getScript();
-                        return new ScriptPair(expectedScript, observedScript);
+                        observedScript = finished.getScript();
+                        
+                        // we keep the result and read any other event is still in the buffer. We will return after the first timeout
+//                        return new ScriptPair(expectedScript, observedScript);
+                        finishedReceived = true;
+                        break;
                     default:
                         throw new IllegalArgumentException("Unrecognized event kind: " + event.getKind());
                     }
                 } catch (SocketTimeoutException e) {
+                    if (finishedReceived)
```

Comment:
Normally, the way a client will implement an AWAIT will be that it will lock a thread waiting for the notification. If this notification will never be sent as a consequence of the already sent FINISHED event, the thread will unlock only with an interrupt (or timeout), which will cause the test to fail, even if it finished on the driver with success.